src/node_mgmt/commands.py

```diff
@@ -55,26 +56,23 @@ def env_scan(node):
     scan_result = scanner.do_scan()
     logger.info('Completed environment scan. Submitting results to API and MQTT')
     node.api.post_env_scan(scan_result)
-    if node.mqtt_client.publish(scan_result, topic=MQTT_STATE_TOPIC):
-        logger.info(f"ENV_SCAN [mqtt]: Successfully pushed")
-    else:
-        # For some reason the env_state wasn't pushed successfully
-        logger.warning(f"ENV_SCAN [mqtt]: Push failed")
 
 
 def trigger_config_generation(node, tank_dimensions=None):
-    logger.info('Starting environment scan')
-    scanner = EnvScanner()
-    scan_result = scanner.do_scan()
-    scan_result[GENERATE_NEW_CONFIG_FLAG] = True
+    logger.info('Starting autoconfig trigger')
+    last_env_scan = None
+    with KVCache() as kvc:
+        last_env_scan = kvc.get(keys.LAST_ENV_SCAN)
+    last_env_scan[GENERATE_NEW_CONFIG_FLAG] = True
```

Comment:
If a scan isn't available in the cache then `last_env_scan` will be `None` and this will raise an exception. In a situation that a scan isn't available it might make sense to simply run `env_scan(node)` here so that one will become available.

Reply:
That's true, though the trigger function is only called after the env_scan so this wouldn't happen now. But I see that this can be used as a command so better to have the env_scan run first

Reply:
Yeah so anything in the `commands.py` module can actually also be triggered via a remote request (by setting the `command` field in Postgres). I would argue that maybe config generation is something that we should exclude from this, but for now it's possible. So we should - at least in theory - consider the possibility that such a remote request would come in, which would then involve a slightly different workflow than the web UI one you've implemented.